File: src/string/string.js

var s = "abcdef";

var grow = function (){
    s= "123" + s + "456"+s + "789";
    s= s.substring( Math.floor(s.length/2),s.length) + s.substring(0, Math.ceil(s.length/2));
    return s;
}

var trial =function (n){
    var i=0;
    do{
        i++;
        grow();
    }while (s.length<=n);
    return s.length;
}

var myTry= function(n){
    var i=0;
    do{
        s="abcdef";
        trial(n);
        i++;
    }while (10 >=i);
    return s.length;
}

myTry(5000000)

File: src/string/string.py
s = "abcdef"

def grow():
    global s
    s = "123" + s + "456" + s + "789"
    half_length = len(s) // 2
    s = s[half_length:] + s[:half_length]
    return s

def trial(n):
    global s
    i = 0
    while len(s) <= n:
        i += 1
        grow()
    return len(s)

def myTry(n):
    global s
    i = 0
    while i < 10:
        s = "abcdef"
        trial(n)
        i += 1
    return len(s)

myTry(5000000)
